File: save.py
#!/usr/bin/python
import os, cgi, cgitb; cgitb.enable()
form = cgi.FieldStorage()
print 'Content-Type: text/plain\n';
if form.has_key('json'):
	json = form['json'].value
	open('state.json', 'wb').write(json)
	print 'saved'
else:
	print 'post.json.plz'

File: save.php
<?
$json = $_POST['json'];
if($json){
	json_decode($json);
	if(json_last_error() == JSON_ERROR_NONE){
		file_put_contents("state.json", $json);
		echo "saved";
	}else{
		echo "json decode error";
	}
}
?>